### ISSUE TITLE
Feature request: VIRTUAL_HOST

### ISSUE THREAD

**Issue by ekkis** [ZZROTDesign/alpine-caddy#27]
if you look at https://github.com/BlackGlory/caddy-proxy you'll see that his package automatically regenerates a Caddyfile every time a new docker container gets fired. it looks at the environment variable, which contains the name we want the container to answer to, and maps appropriately.

it'd be great if this package could do this. reason: this package has 100+K pulls whilst BlackGlory's has 385 and gets no support. I can't make it work (I've already spent too much time trying) but I love the functionality

would it be difficult to do?

**Comment by kilpatty** [comment]
So just to be clear, this would have caddy act as a reverse proxy that can automatically detect new services?

That may be out of the scope of this project, but I'm sure there is some way that we can fit it in whether it's a separate branch, or a fork from this repo that I wouldn't mind maintaining.

**Comment by ekkis** [comment]
yes, apparently https://github.com/jwilder/nginx-proxy does it and it was the inspiration for BlackGlory's image
